doc/upgrade.md

```diff
@@ -0,0 +1,138 @@
+
+
+## Problem 2: device not found in inst-sys
+
+The second problem comes when the mounting action is peformed by using the encryption name read from the fstab file. As mentioned above, the fstab entry for an encryption device could be one of the following:
+
+```
+UUID=111-222-333                            /home   ext4    defaults 0 0  # mount_by = uuid
+/dev/mapper/cr_ata-111-part2                /home   ext4    defaults 0 0  # mount_by = device name
+LABEL=home_enc                              /home   ext4    defaults 0 0  # mount_by = volume label
+/dev/disk/by-id/dm-name-cr_ata-111-part2    /home   ext4    defaults 0 0
+/dev/disk/by-uuid/111-222-333               /home   ext4    defaults 0 0
+/dev/disk/by-label/home_enc                 /home   ext4    defaults 0 0
+```
+
+The fstab spec could be problematic when it includes the encryption device name as part of the device spec, that is:
+
+```
+/dev/mapper/cr_ata-111-part2                /home   ext4    defaults 0 0
+/dev/disk/by-id/dm-name-cr_ata-111-part2    /home   ext4    defaults 0 0
+```
+
+In such cases, the mount command would fail if the encryption device was probed with another name. For example, if the encryption device was probed with "cr-auto-1" name, the inst-sys will contain the following device names:
+
+```
+/dev/mapper/cr-auto-1
+/dev/disk/by-id/dm-name-cr-auto-1
+```
+
+So, when the fstab device tryied to be mounted, it will fail:
+
+```
+mount /dev/mapper/cr_ata-111-part2 /home  => fail device /dev/mapper/cr_ata-111-part2 does not exist
+```
+
+The solution here is to avoid to use the fstab spec when that device name is not safe enough. So, instead of using the fstab spec, the udev uuid name of the encryption device will be used due to that name exists in the inst-sys:
+
+```
+mount /dev/disk/by-uuid/111-222-333 /home
+```
+
+Note that this problem could not appear when upgrading a system generated with old storage. The old storage created the fstab entries with the option "nofail" for some mount points, for example "/" or "/home":
+
+Note that the old storage was creating fstab entries with the option "nofail" for some mount points, for example "/" or "/home". Therefore, a system generated with old storage could silently fail when trying to mount the filesystem, and the upgrade process could finish without that volume being mounted.
+
+```
+/dev/mapper/cr_ata-111-part2                /home   ext4    nofail 0 0
+```
+
+This could make the upgrade useless when some important mount point is not properly mounted, for example, when mounting an encrypted /usr silently fails.
+
+Those problems are now fixed for SLE-15-SP1, see Pull Requests for [libstorage-ng](https://github.com/openSUSE/libstorage-ng/pull/589), [yast2-storage-ng](https://github.com/yast/yast-storage-ng/pull/790) and [yast2-update](https://github.com/yast/yast-update/pull/115).
```

Comment:
NP (grammar Nazi): same again here. Upper casing "Pull Requests" doesn't seem to make much sense. The fact that you can use it in singular and plural proves it's not a proper name.